faster_whisper/transcribe.py

```diff
@@ -1132,51 +1811,350 @@ def find_alignment(
+                    )
+                ]
+            )
+        return return_list
+
+    def generate_segment_batched(
+        self,
+        features: torch.Tensor,
+        tokenizer: Tokenizer,
+        options: dict,
+    ):
+        batch_size = features.shape[0]
+        all_tokens = []
+        prompt_reset_since = 0
 
-        words, word_tokens = tokenizer.split_to_word_tokens(
-            text_tokens + [tokenizer.eot]
+        if options["initial_prompt"] is not None:
+            initial_prompt = " " + options["initial_prompt"].strip()
+            initial_prompt_tokens = tokenizer.encode(initial_prompt)
+            all_tokens.extend(initial_prompt_tokens)
+        previous_tokens = all_tokens[prompt_reset_since:]
+        prompt = self.get_prompt(
+            tokenizer,
+            previous_tokens,
+            without_timestamps=options["without_timestamps"],
+            prefix=options["prefix"],
         )
-        if len(word_tokens) <= 1:
-            # return on eot only
-            # >>> np.pad([], (1, 0))
-            # array([0.])
-            # This results in crashes when we lookup jump_times with float, like
-            # IndexError: arrays used as indices must be of integer (or boolean) type
-            return []
-        word_boundaries = np.pad(np.cumsum([len(t) for t in word_tokens[:-1]]), (1, 0))
-        if len(word_boundaries) <= 1:
-            return []
 
-        jumps = np.pad(np.diff(text_indices), (1, 0), constant_values=1).astype(bool)
-        jump_times = time_indices[jumps] / self.tokens_per_second
-        start_times = jump_times[word_boundaries[:-1]]
-        end_times = jump_times[word_boundaries[1:]]
-        word_probabilities = [
-            np.mean(text_token_probs[i:j])
-            for i, j in zip(word_boundaries[:-1], word_boundaries[1:])
+        encoder_output = self.encode(features)
+
+        result = self.model.generate(
+            encoder_output,
+            [prompt] * batch_size,
+            beam_size=options["beam_size"],
+            patience=options["patience"],
+            length_penalty=options["length_penalty"],
+            max_length=self.max_length,
+            suppress_blank=options["suppress_blank"],
+            suppress_tokens=options["suppress_tokens"],
+            return_scores=True,
+            return_no_speech_prob=True,
+        )
+
+        output = []
+        for res in result:
+            output.append({})
+            # return scores
+            seq_len = len(res.sequences_ids[0])
+            cum_logprob = res.scores[0] * (seq_len ** options["length_penalty"])
+            output[-1]["avg_logprob"] = cum_logprob / (seq_len + 1)
+
+            # return no speech prob
+            output[-1]["no_speech_prob"] = res.no_speech_prob
+            output[-1]["tokens"] = res.sequences_ids[0]
+
+        return encoder_output, output
+
+    def detect_language_function(self, audio: torch.Tensor):
```

Comment:
Duplicate code between detect_language and detect_language_function

Reply:
this is `detect_language` function but it was renamed, although I'm in favor of the old name, and if the reason for rename was to avoid confusion with `ctranslate2.models.Whisper.detect_language`, I think it won't be a problem because we have `WhisperModel.encode` and `ctranslate2.models.Whisper.encode`

Reply:
It was renamed to cater to "unused function" formatting error with `flake8`:

`./faster_whisper/transcribe.py:1937:5: F811 redefinition of unused 'detect_language' from line 1922`

Reply:
I think you should remove function `detect_language_function`. And change the logic as below:
```python
  (
      language,
      language_probability,
      all_language_probs,
  ) = self.model.detect_language(audio)
 ```
 I checked and there is no flake8 error

Reply:
I also got no flake8 errors before the change